source/iterator/worker.go

```diff
@@ -0,0 +1,350 @@
+// Copyright © 2022 Meroxa, Inc. & Yalantis
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package iterator
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/url"
+	"time"
+
+	"github.com/conduitio-labs/conduit-connector-redshift/columntypes"
+	"github.com/conduitio/conduit-commons/opencdc"
+	sdk "github.com/conduitio/conduit-connector-sdk"
+	"github.com/huandu/go-sqlbuilder"
+	"github.com/jmoiron/sqlx"
+)
+
+const (
+	// keySearchPath is a key of get parameter of a datatable's schema name.
+	keySearchPath = "search_path"
+	// pingTimeout is a database ping timeout.
+	pingTimeout = 10 * time.Second
+)
+
+type Worker struct {
+	db       *sqlx.DB
+	rows     *sqlx.Rows
+	position TablePosition
+
+	// table is a table name
+	table string
+	// keyColumns are table column which the worker will use to create a record's key
+	keyColumns []string
+	// orderingColumn is the name of the column that worker will use for sorting data
+	orderingColumn string
+	// batchSize is the size of a batch retrieved from Redshift
+	batchSize int
+	// columnTypes is a mapping from column names to their respective types
+	columnTypes map[string]string
+
+	// iterator is an instance of the iterator
+	iterator *Iterator
```

Comment:
Hmm, what's `Worker` supposed to do, and what is `Iterator` supposed to do? IIUC, `Iterator` is a "global" iterator, for the whole source connector, and `Worker` is actually an iterator for a single table?

Reply:
Yes right

Reply:
In that case, I'd like to suggest to rename so that it's clear. From the name itself it's not clear that a Worker is actually an iterator (only after going through the code that becomes clear).

Reply:
Sure, I'll rename it to "tableIterator" maybe.